src/codewhisperer/service/keyStrokeHandler.ts

```diff
@@ -190,6 +192,17 @@ export class KeyStrokeHandler {
                 } finally {
                     RecommendationHandler.instance.isGenerateRecommendationInProgress = false
                 }
+            } else if (isInlineCompletionEnabled()) {
+                if (!vsCodeState.isCodeWhispererEditing && !InlineCompletionService.instance.isPaginationRunning()) {
```

Comment:
Usually it's better to hide these types of guards within the implementation rather than making it the caller's responsibility.

Reply:
Agree. Will move it inside.